File: source/java/workshop/company-service/src/main/java/org/workshop/companyservice/service/CompanyService.java
package org.workshop.companyservice.service;

import org.workshop.companyservice.dto.CompanyList;
import org.workshop.companyservice.entity.Company;

import java.util.UUID;

public interface CompanyService {
    Company save(Company company);
    Company findById(UUID uuid);
    CompanyList findAll();
    Company update(UUID id, Company company) throws Exception;
    void delete(UUID uuid) throws Exception;
}

File: source/angular/workshop-angular/src/app/companies/shared/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { CompanyList } from './company-list.model';
import { Company } from "./company.model";
import { environment } from '../../../environments/environment';

// const httpOptions = {
//   headers: new HttpHeaders ({
//     "Access-Control-Allow-Origin": "**"
//   })
// }


@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<CompanyList> {
    return this.http
      .get<CompanyList>(
        environment.company_service_url
      )
      .pipe(map((company) => company || {}));
  }


  save(company: Company): Observable<any> {
    return this.http
      .post<any>(
        environment.company_service_url,
        company
      );

  }

  findById() {

  }

  update() {

  }

  delete() {

  }

}
